newdle/client/src/Timeline.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import Participant from './Participant';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import {Header} from 'semantic-ui-react';
+import styles from './Timeline.module.scss';
+import moment from 'moment';
+
+function calculateWidth(start, end, minHour, maxHour) {
+  let startMins = start.hours() * 60 + start.minutes();
+  let endMins = end.hours() * 60 + end.minutes();
+
+  if (startMins < minHour * 60) {
+    startMins = minHour * 60;
+  }
+  if (endMins > maxHour * 60) {
+    endMins = maxHour * 60;
+  }
+
+  return ((endMins - startMins) / ((maxHour - minHour) * 60)) * 100;
+}
+
+function calculatePosition(start, minHour, maxHour) {
+  const spanMins = (maxHour - minHour) * 60;
+  let startMins = start.hours() * 60 + start.minutes() - minHour * 60;
+
+  if (startMins < 0) {
+    startMins = 0;
+  } else if (startMins >= spanMins) {
+    startMins = spanMins - 5;
```

Comment:
Why are you subtracting 5 here? Is it because of some margin/padding?

Reply:
It's the case when start time is later than the end time of the timeline. We are subtracting 5 from end time so that the row is visible

Reply:
Ok, but does it have to be 5?

Reply:
I understand your point now, I've changed the way we render the overflowing slots